torch/testing/_internal/common_nn.py

```diff
@@ -5072,7 +5085,8 @@ def _do_test(self, test_case, module, input):
             test_case.assertNotEqual(input_ip_clone._version, input_version)
             test_case.assertEqual(output, output_ip)
             grad = output.data.clone().normal_()
-            input.grad.data.zero_()
+            if input.grad is not None:
+                input.grad.data.zero_()
```

Comment:
nit: we should use `torch.no_grad()` or `detach()` instead of `.data`. Not introduced in this PR though so fine if you don't fix it.

Reply:
Good point,  I will update (or at least try to and see if anything fails)